src/Avalonia.Controls/ApplicationLifetimes/ClassicDesktopStyleApplicationLifetime.cs

```diff
@@ -122,12 +122,20 @@ public int Start(string[] args)
                 lifetimeEvents.ShutdownRequested += OnShutdownRequested;
 
             _cts = new CancellationTokenSource();
-            MainWindow?.Show();
+            ShowMainWindow(); // Note due to a bug in the JIT we wrap this in a method, otherwise MainWindow
+                              // gets stuffed into a local var and can not be GCed until after the program stops.
+                              // this method never exits until program end.
+                              
             Dispatcher.UIThread.MainLoop(_cts.Token);
             Environment.ExitCode = _exitCode;
             return _exitCode;
         }
 
+        private void ShowMainWindow()
```

Comment:
Shouldn't this method have `[MethodImpl(MethodImplOptions.NoInlining)]` otherwise we might end up with a local var again, don't we?

Reply:
done.